### PR TITLE
Added Object3D.onUpdate() method

### PR DESCRIPTION
**Pull request by WestLangley**
As discussed in #11635.

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

**Comment by WestLangley** [comment]
@mrdoob Look OK?

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1108,6 +1108,14 @@ function WebGLRenderer( parameters ) {
 		_currentMaterialId = - 1;
 		_currentCamera = null;
 
+		// onUpdate
+
+		scene.traverse( function( node ) {
+
+			node.onUpdate( _this, scene, camera );
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: How about calling this in the `projectObject` function? That way we do a single scene traverse...

Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: I was thinking it should be before the world matrices are calculated...

Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: There are tradeoffs in either case. Go ahead and merge or modify to your liking. I have a PR for the light helpers in the queue.

If you are having second thoughts about this, that is OK...

Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: Ah... Good point... Hmm... So tempting to just putting it in `updateMatrixWorld()`...

Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: Time for an executive decision... :)

Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: Could we have a way of registering objects for which `onUpdate()` needs to be called? That way, the renderer just needs to maintain a short array.

**Comment by WestLangley** [comment]
Maybe someone else can propose a different implementation, then...

It still would be nice to support callbacks for non-renderable objects -- such as the helpers, or a model's root object.

Closing.

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: > Time for an executive decision... :)

Yep! 😅

**Comment by mrdoob** [comment]
@WestLangley Just realised that we can (ab)use `updateMatrixWorld` to update the helpers automatically 😊

**Comment by WestLangley** [comment]
@mrdoob OK...

**Pull request closed without merging**